src/Catalyst.Core.Modules.Dfs/Migration/MigrationManager.cs

```diff
@@ -0,0 +1,147 @@
+        ///   for the specifed <see cref="DfsService"/>.
+        /// </summary>
+        public MigrationManager(RepositoryOptions options)
+        {
+            _options = options;
+            Migrations = typeof(MigrationManager).Assembly.GetTypes()
+               .Where(x => typeof(IMigration).IsAssignableFrom(x) && !x.IsInterface && !x.IsAbstract)
+               .Select(x => (IMigration) Activator.CreateInstance(x))
+               .OrderBy(x => x.Version)
+               .ToList();
+        }
+
+        /// <summary>
+        ///   The list of migrations that can be performed.
+        /// </summary>
+        public List<IMigration> Migrations { get; }
+
+        /// <summary>
+        ///   Gets the latest supported version number of a repository.
+        /// </summary>
+        public int LatestVersion => Migrations.Last().Version;
+
+        /// <summary>
+        ///   Gets the current vesion number of the  repository.
+        /// </summary>
+        public int CurrentVersion
+        {
+            get
+            {
+                var path = VersionPath();
+                if (!File.Exists(path))
+                {
+                    return 0;
+                }
+                
+                using (var reader = new StreamReader(path))
+                {
+                    var s = reader.ReadLine();
+                    return int.Parse(s ?? throw new NullReferenceException(("stream null")));
+                }
+            }
+            private set => File.WriteAllText(VersionPath(), value.ToString(CultureInfo.InvariantCulture));
+        }
+
+        /// <summary>
+        ///   Upgrade/downgrade to the specified version.
+        /// </summary>
+        /// <param name="version">
+        ///   The required version of the repository.
+        /// </param>
+        /// <param name="cancel">
+        /// </param>
+        /// <returns></returns>
+        public async Task MirgrateToVersionAsync(int version, CancellationToken cancel = default)
+        {
+            if (version != 0 && Migrations.All(m => m.Version != version))
+            {
+                throw new ArgumentOutOfRangeException(nameof(version), $@"Repository version '{version.ToString()}' is unknown.");
+            }
+
+            var currentVersion = CurrentVersion;
+            var increment = CurrentVersion < version ? 1 : -1;
+            while (currentVersion != version)
+            {
+                var nextVersion = currentVersion + increment;
+                Log.InfoFormat("Migrating to version {0}", nextVersion.ToString());
+
+                if (increment > 0)
+                {
+                    var migration = Migrations.FirstOrDefault(m => m.Version == nextVersion);
+                    if (migration != null && migration.CanUpgrade)
+                    {
+                        await migration.UpgradeAsync(_options, cancel);
+                    }
+                }
+                else if (increment < 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add the missing 'else' clause.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)